samples/ingestion/ingestion-client/Connector/StorageConnector.cs

```diff
@@ -10,104 +10,51 @@ namespace Connector
     using System.Linq;
     using System.Text;
     using System.Threading.Tasks;
-    using System.Web;
     using Azure.Storage;
     using Azure.Storage.Blobs;
     using Azure.Storage.Sas;
     using Microsoft.Extensions.Logging;
 
-    public class StorageConnector
+    public class StorageConnector : IStorageConnector
     {
         private readonly BlobServiceClient blobServiceClient;
+        private readonly StorageSharedKeyCredential storageCredential;
+        private readonly ILogger<StorageConnector> log;
 
-        private readonly string accountName;
-
-        private readonly string accountKey;
-
-        public StorageConnector(string storageConnectionString)
+        public StorageConnector(BlobServiceClient blobServiceClient, StorageSharedKeyCredential storageCredential, ILogger<StorageConnector> log)
         {
-            if (storageConnectionString == null)
-            {
-                throw new ArgumentNullException(nameof(storageConnectionString));
-            }
-
-            this.accountName = GetValueFromConnectionString("AccountName", storageConnectionString);
-            this.accountKey = GetValueFromConnectionString("AccountKey", storageConnectionString);
-
-            this.blobServiceClient = new BlobServiceClient(storageConnectionString);
+            this.blobServiceClient = blobServiceClient ?? throw new ArgumentNullException(nameof(blobServiceClient));
+            this.storageCredential = storageCredential ?? throw new ArgumentNullException(nameof(storageCredential));
+            this.log = log ?? throw new ArgumentNullException(nameof(log));
         }
 
-        public static string GetFileNameFromUri(Uri fileUri)
+        public string GetFileNameFromUri(Uri fileUri)
         {
             if (fileUri == null)
             {
                 throw new ArgumentNullException(nameof(fileUri));
             }
 
-            var pathParts = HttpUtility.UrlDecode(fileUri.AbsolutePath).Split('/').ToList();
+            var pathParts = System.Web.HttpUtility.UrlDecode(fileUri.AbsolutePath).Split('/').ToList();
```

Comment:
nit: why don't you just keep the 'using System.Web;' ?

Reply:
Updated